Source/Dafny/Resolver.cs

```diff
@@ -6838,7 +6838,11 @@ public CheckTypeInference_Visitor(Resolver resolver, ICodeContext codeContext)
           }
         } else if (CheckTypeIsDetermined(expr.tok, expr.Type, "expression")) {
           if (expr is UnaryOpExpr uop) {
-            uop.ResolveOp(); // Force resolution eagerly at this point to catch potential bugs
+            // The CheckTypeInference_Visitor has already visited uop.E, but uop.E's may be undetermined. If that happened,
+            // then an error has already been reported.
+            if (CheckTypeIsDetermined(uop.E.tok, uop.E.Type, "expression")) {
+              uop.ResolveOp(); // Force resolution eagerly at this point to catch potential bugs
+            }
```

Comment:
Are you sure about this part?  resolution for unary ops is lazy, as opposed to binary ops that are resolved explicitly when the resolver traverses the AST.

Reply:
Frankly, I'm _not_ sure about this change. It should be fine like it is now, for the reason explain in the comment on lines 6841-6842. The thing I'm not sure about is why the resolver started crashing here when I made the other changes. The `uop.ResolveOp()` method expects that `uop.E` has a legal type (and it crashes if it detects otherwise). I tried to provoke this crash in the previous version of Dafny, but wasn't able to. I also tried to see how the other changes I implemented could cause a change that brings control here, but was stumped.

So, I feel good about the new code here, but I don't understand why this wasn't a problem before.